contracts/zero-ex/contracts/src/features/UniswapV3Feature.sol

```diff
@@ -175,6 +176,33 @@ contract UniswapV3Feature is
         );
     }
 
+    /// @dev Sell a token for another token directly against uniswap v3.
+    ///      Private variant, uses tokens held by `address(this)`.
+    /// @param encodedPath Uniswap-encoded path.
+    /// @param sellAmount amount of the first token in the path to sell.
+    /// @param minBuyAmount Minimum amount of the last token in the path to buy.
+    /// @param recipient The recipient of the bought tokens. Can be zero for sender.
+    /// @return buyAmount Amount of the last token in the path bought.
+    function _sellTokenForTokenToUniswapV3(
```

Comment:
I thought about it but figured it would be sketchier security-wise (unless we were to enforce in the function that `payer` is equal to `msg.sender` or `address(this)`)

Reply:
I guess that would make MultiplexUniswapV3 slightly cleaner 🤔

Reply:
I think we already assume the same level of risk when calling the internal `_transformERC20()` fn. We can highlight these spots for the auditor. But I think it should be somewhat easy to prove that payer will always be either the taker or the EP, which is all we care about?

Reply:
the security assumption would the same, but I'd prefer to minimize the number of functions that make that security assumption. Probably won't matter, but imo the surface area for a potential vulnerability is slightly smaller this way